src/components/tx/DecodedTx/index.tsx

```diff
@@ -1,44 +1,64 @@
 import { type ReactElement } from 'react'
 import { Accordion, AccordionDetails, AccordionSummary, Box, Skeleton } from '@mui/material'
 import { type SafeTransaction } from '@gnosis.pm/safe-core-sdk-types'
-import { type DecodedDataResponse, getDecodedData } from '@gnosis.pm/safe-react-gateway-sdk'
+import {
+  type DecodedDataResponse,
+  getDecodedData,
+  getTransactionDetails,
+  type TransactionDetails,
+} from '@gnosis.pm/safe-react-gateway-sdk'
 import useChainId from '@/hooks/useChainId'
 import useAsync from '@/hooks/useAsync'
 import { MethodDetails } from '@/components/transactions/TxDetails/TxData/DecodedData/MethodDetails'
 import ErrorMessage from '../ErrorMessage'
-import { generateDataRowValue } from '@/components/transactions/TxDetails/Summary/TxDataRow'
+import Summary from '@/components/transactions/TxDetails/Summary'
 
 type DecodedTxProps = {
   tx?: SafeTransaction
+  txId?: string
 }
 
-const DecodedTx = ({ tx }: DecodedTxProps): ReactElement | null => {
+const DecodedTx = ({ tx, txId }: DecodedTxProps): ReactElement | null => {
   const chainId = useChainId()
   const encodedData = tx?.data.data
   const isNativeTransfer = encodedData && isNaN(parseInt(encodedData, 16))
 
-  const [decodedData, error] = useAsync<DecodedDataResponse>(() => {
+  const [decodedData, decodedDataError, decodedDataLoading] = useAsync<DecodedDataResponse>(() => {
     if (!encodedData || isNativeTransfer) return
     return getDecodedData(chainId, encodedData)
   }, [chainId, encodedData, isNativeTransfer])
 
-  if (isNativeTransfer) return null
+  const [txDetails, txDetailsError, txDetailsLoading] = useAsync<TransactionDetails>(() => {
+    if (!txId) return
+    return getTransactionDetails(chainId, txId)
+  }, [])
+
+  if (isNativeTransfer && !txId) {
+    return null
+  }
 
   return (
     <Box mb={2}>
       <Accordion elevation={0}>
-        <AccordionSummary>
-          <Box flex={1}>Transaction data</Box>
-          {encodedData ? generateDataRowValue(encodedData, 'rawData') : ''}
-        </AccordionSummary>
+        <AccordionSummary>Transaction details</AccordionSummary>
 
         <AccordionDetails>
+          {txDetails ? (
+            <Box mb={1}>
+              <Summary txDetails={txDetails} defaultExpanded />
+            </Box>
+          ) : txDetailsError ? (
+            <ErrorMessage error={txDetailsError}>Failed loading transaction details</ErrorMessage>
+          ) : (
+            txDetailsLoading && <Skeleton />
+          )}
+
           {decodedData ? (
             <MethodDetails data={decodedData} />
-          ) : error ? (
-            <ErrorMessage error={error}>Failed decoding transaction data</ErrorMessage>
+          ) : decodedDataError ? (
+            <ErrorMessage error={decodedDataError}>Failed decoding transaction data</ErrorMessage>
           ) : (
-            <Skeleton />
+            decodedDataLoading && <Skeleton />
```

Comment:
Should we display this part above the details since the method is more general than the tx details?

Reply:
I put it below because the last item in tx details is "raw data", so it kinda continues it nicely.

Reply:
Also I don't see it as more general. All txns have a safeTxHash, but not all have any tx data (e.g. native transfers don't).